### ISSUE TITLE
Provide a way to reference the project directory in additional_dependencies

### ISSUE THREAD

**Issue by ulope** [pre-commit/pre-commit#1406]
Currently it seems not possible to reference anything from inside the project directory in `additional_dependencies` since (in case of Python) the install command is run in the context of the current hook.

This example
```
        additional_dependencies: ["-r", "requirements/_pip-tools.txt"]
```
leads to
```
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements/_pip-tools.txt'
```

It would be very useful if there was a way to either 
- change the working directory of the `pip install` call
- provide a variable / placeholder that points to the project directory that can be used in the `additional_dependencies`

**Comment by asottile** [comment]
#730 has some more information on this, but the long and the short of it is this cannot and will not be supported: https://github.com/pre-commit/pre-commit/issues/730#issuecomment-542269775

**Comment by ulope** [comment]
Ok I understand the general difficulty. 

I've solved the general caching problem for our use case with a [custom wrapper script](https://github.com/raiden-network/raiden/blob/10f98693f4bf5edce88d1e7f43ae294c7f6b084a/tools/pre-commit/pre-commit-wrapper.py) that updates the hook venv if the hash of the requirements.txt changes. 

The only thing that is still problematic is that the pip-tools version referenced in `.pre-commit-config.yaml` tends to get forgotten to be updated and therefore goes out of sync with the main repo.

**Comment by asottile** [comment]
our solution at lyft is to synchronize the two -- we do it via an external wrapper but it could be done via a script or even a pre-commit hook -- there's open work to make something like this available publicly and more generically in #945
